### PR TITLE
Docs: Update Vite docs to v7

### PR DESCRIPTION
**Pull request by JReinhold**
## What I did

Minor changes to the Vite docs:

1. Describe how Vite now automatically merges configuration with the default `vite.config.js`
2. Remove section about `svelteOptions` as that is no longer supported
3. Included Svelte in list of frameworks with arg type inference.

There's still more work to be done, that I've left out of this PR to move this forward quickly:

1. Update builder docs to no longer mention `@storybook/builder-vite` as that is configured with the framework instead. this is also applicable for the Webpack docs.
2. Describe that you can customize the path to your Vite config. introduced in https://github.com/storybookjs/storybook/pull/20681, described at https://github.com/storybookjs/storybook/tree/next/code/lib/builder-vite#customize-vite-config

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: docs/builders/vite.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ Update your Storybook configuration (in `.storybook/main.js|ts`) to include the
 
 ## Configuration
 
-Out of the box, Storybook's Vite builder includes a set of configuration defaults for the supported frameworks. You can also fine-tune them or override them to match your existing configuration as, by default, the builder does not read your `vite.config.js` file. For example, if you need to set up aliasing, you can adjust your Storybook configuration file (`.storybook/main.js|ts`) and provide the following:
+Out of the box, Storybook's Vite builder includes a set of configuration defaults for the supported frameworks, merged with your default `vite.config.js` configuration. We recommend keeping your Storybook Vite configuration the same as your app Vite configuration by only making changes to your `vite.config.js`. But you can also fine-tune the configuration specifically for Storybook by adjusting your Storybook configuration file (`.storybook/main.js|ts`) and provide the following:
```
Comment (IanVS) [inline docs/builders/vite.md:41]:
```suggestion
Out of the box, Storybook's Vite builder includes a set of configuration defaults for the supported frameworks, merged with your `vite.config.js` configuration file. We recommend keeping your Storybook Vite configuration the same as your app Vite configuration by only making changes to your `vite.config.js`. But you can also fine-tune the configuration specifically for Storybook by adjusting your Storybook configuration file (`.storybook/main.js|ts`) and providing the following:
```

File: docs/snippets/common/storybook-vite-builder-aliasing.js.mdx [hunk 2]
```diff
@@ -12,8 +12,6 @@ export default {
   async viteFinal(config) {
     // Merge custom configuration into the default config
     return mergeConfig(config, {
-      // Use the same "resolve" configuration as your app
-      resolve: (await import('../vite.config.js')).default.resolve,
       // Add dependencies to pre-optimization
       optimizeDeps: {
         include: ['storybook-dark-mode'],
```
Comment (IanVS) [inline docs/snippets/common/storybook-vite-builder-aliasing.js.mdx:17]: We might want to pick a different addon, as far as I can tell this one is unneeded in v7 and is not compatible either.

**Review by IanVS — APPROVED** [review]
Thanks for the cleanup, I could have sworn I updated these earlier, but maybe that was just in the README file.  I had a few small suggestions, but no blockers.

File: docs/builders/vite.md [hunk 3]
```diff
@@ -50,7 +50,7 @@ Out of the box, Storybook's Vite builder includes a set of configuration default
 
 <!-- prettier-ignore-end -->
 
-The asynchronous function`viteFinal` receives a `config` object with the default builder configuration and returns the updated configuration with the defined alias.
+The asynchronous function`viteFinal` receives a `config` object with the default builder configuration and returns the updated configuration.
```
Comment (kylegach) [inline docs/builders/vite.md:53]:
```suggestion
The asynchronous function `viteFinal` receives a `config` object with the default builder configuration and returns the updated configuration.
```

**Review by kylegach — APPROVED** [review]
Looks like there's some minor comments, but I'll go ahead and approve.

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@JReinhold when you're able can you apply the change I've mentioned? Thanks in advance.

File: docs/builders/vite.md [hunk 4]
```diff
@@ -38,7 +38,7 @@ Update your Storybook configuration (in `.storybook/main.js|ts`) to include the
 
 ## Configuration
 
-Out of the box, Storybook's Vite builder includes a set of configuration defaults for the supported frameworks. You can also fine-tune them or override them to match your existing configuration as, by default, the builder does not read your `vite.config.js` file. For example, if you need to set up aliasing, you can adjust your Storybook configuration file (`.storybook/main.js|ts`) and provide the following:
+Out of the box, Storybook's Vite builder includes a set of configuration defaults for the supported frameworks, merged with your `vite.config.js` configuration file. We recommend keeping your Storybook Vite configuration the same as your app Vite configuration by only making changes to your `vite.config.js`. But you can also fine-tune the configuration specifically for Storybook by adjusting your Storybook configuration file (`.storybook/main.js|ts`) and providing the following:
```
[the remark this answers is not in the document]
Comment (JReinhold) [reply docs/builders/vite.md]: Thanks. I changed it to something similar, but wording that doesn't talk about configuration _changes_, but rather configuration in general, changed or static. I don't think it makes sense to think of them as changes that needs to be picked up.

**Review by jonniebigodes — APPROVED** [review]
@JReinhold, thanks for addressing the feedback so promptly. Appreciate it. Regarding your comment, we can always circle back to this and update it later on. Going to approve this to unblock you.

**Pull request merged**